### PR TITLE
Crusher improvements

### LINKED ISSUE #8300: [BUG] Crusher hut page 2 title missing
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Are you using the latest MineColonies Verison?

- [X] I am running the latest **alpha** version of MineColonies for my Minecraft version.

### Did you check on the Wiki? or ask on Discord?

- [X] I checked the MineColonies Wiki and made sure my issue is not covered there. Or I was sent from discord to open an issue here.

### Minecraft Version

1.18

### MineColonies Version

1.18.2-1.0.659-ALPHA

### Structurize Version

1.18.2-1.0.389-RELEASE

### Related Mods and their Versions

_No response_

### Current Behavior

The first page of the Crusher hut GUI still has the old blue arrow interface, which leads to page 2. This is where you can select what the hut can crush.  The title of the page says "MISSING"
crushergui2

### Expected Behavior

Change the page to the new tab interface and give it a title.

### Reproduction Steps

Open Crusher Hut GUI and go to page 2

### Logs

no log needed

### Anything else?

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

### LINKED ISSUE #8304: [BUG] Flower hut can't harvest ferns properly
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Are you using the latest MineColonies Verison?

- [X] I am running the latest **alpha** version of MineColonies for my Minecraft version.

### Did you check on the Wiki? or ask on Discord?

- [X] I checked the MineColonies Wiki and made sure my issue is not covered there. Or I was sent from discord to open an issue here.

### Minecraft Version

1.18

### MineColonies Version

1.18.2-1.0.662

### Structurize Version

1.18.2-1.0.389

### Related Mods and their Versions

The minecolonies depedencies listed on the file download page for both 1.0.625 and 1.0.662, forge 40.1.0 and a few earlier versions, immersive engineering 1.18.2-8.0.0146 and later 8.0.1-147, little logistics 1.2.1 and later 1.2.3 (though I doubt these would interact?)

### Current Behavior

1. The flower hut is successfully growing ferns and tall ferns
2. The level 5  flower hut has requested and been provided with both an iron axe,, and later a silk touch enchanted diamond axe
3. The flower hut is not requesting any other tools
4. When the worker 'harvests' ferns or tall ferns, they are simply broken, and yield either nothing, or grass seeds

### Expected Behavior

Should yield ferns/tall ferns as drops

### Reproduction Steps

create a flower hut and have them grow ferns/tall ferns and harvest them with the axe

### Logs

null

### Anything else?

Can test later with no extra mods if desired

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

### LINKED ISSUE #8308: [BUG] <title>Couriers stop working
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Are you using the latest MineColonies Verison?

- [X] I am running the latest **alpha** version of MineColonies for my Minecraft version.

### Did you check on the Wiki? or ask on Discord?

- [X] I checked the MineColonies Wiki and made sure my issue is not covered there. Or I was sent from discord to open an issue here.

### Minecraft Version

1.16

### MineColonies Version

664

### Structurize Version

392

### Related Mods and their Versions

Forge Version: 36.2.34

### Current Behavior

 Since a few days (irl) i have the problem that my couriers stop working every now and than. They just stay in their assigned warehouse until they go to bed. All the common fixes (punching them, restarting, even recruiting an other person) didnt work. I have no problems with any other job. The bug hast started to appear in the previous release version i think
Resetting the request system helps for only a couple of minutes, then they go back idling in the warehouse

### Expected Behavior

To not idle and do their job 

### Reproduction Steps

it just happens after a few minutes for me nothing out of the regular 

### Logs

https://gist.github.com/MockingSniper/13e1f5b723b13cef96a22055eef4f035

### Anything else?

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #8300
Closes #8304
Closes #8308

# Changes proposed in this pull request:
- Add a new Setting type: Recipe Settings, allows to set certain working modes based on recipes (we use it for the crafter herE)
- Remove some old UI code of the sifter.
- Hide sifter recipe tab if no recipes are unlocked yet
- Adjust a few messed up lang strings
- Florist use shears for the right blocks

Review please

### CODE REVIEW

**Comment by uecasm** [comment]
> Hide sifter recipe tab if no recipes are unlocked yet

When would this be the case?  As soon as you build the level 1 sifter they should have string-mesh-based recipes already...

**Comment by Raycoms** [comment]
> > Hide sifter recipe tab if no recipes are unlocked yet
> 
> When would this be the case? As soon as you build the level 1 sifter they should have string-mesh-based recipes already...

At level 0 =D

File: src/api/java/com/minecolonies/api/colony/buildings/modules/settings/ICraftingSetting.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+     * Get the setting value.
+     * @return the current value.
+     */
+    IToken<?> getValue(final IBuilding building);
+
+    /**
+     * Get the setting value.
+     * @return the current value.
+     */
+    IRecipeStorage getValue(final IBuildingView building);
```
Comment (uecasm) [inline src/api/java/com/minecolonies/api/colony/buildings/modules/settings/ICraftingSetting.java:25]: Seems a bit weird this has different return types between client/server interfaces...  and inconsistent with `set`

**Comment by Raycoms** [comment]
> > > Hide sifter recipe tab if no recipes are unlocked yet
> > 
> > 
> > When would this be the case? As soon as you build the level 1 sifter they should have string-mesh-based recipes already...
> 
> At level 0 =D

I'll also add that to the smeltery for the port. It confuses players to see empty lists they can't add anything to

File: src/api/java/com/minecolonies/api/colony/buildings/modules/settings/ICraftingSetting.java (hunk 1 above)
Comment (Raycoms) [reply src/api/java/com/minecolonies/api/colony/buildings/modules/settings/ICraftingSetting.java:25]: I agree, let me standardize that

**Comment by uecasm** [comment]
> I'll also add that to the smeltery for the port. It confuses players to see empty lists they can't add anything to

Should that apply to L1/2 restaurant too?  The convention there was just to hide the Teach button when under-level (although there was a bug and this wasn't happening until fixed recently).  This could also produce a blank tab.

**Comment by Raycoms** [comment]
> > I'll also add that to the smeltery for the port. It confuses players to see empty lists they can't add anything to
> 
> Should that apply to L1/2 restaurant too? The convention there was just to hide the Teach button when under-level (although there was a bug and this wasn't happening until fixed recently). This could also produce a blank tab.

Generally, i'd say: If you can't teach anything and the list is empty, don't display

**Comment by uecasm** [comment]
> Generally, i'd say: If you can't teach anything and the list is empty, don't display

So, something like:
```
    @Override
    public boolean isPageVisible()
    {
        return isVisible && (isRecipeAlterationAllowed() || getRecipes().size() > 0);
    }
```

Or perhaps the equivalent on the server side instead, to calculate `isVisible`.

**Comment by Raycoms** [comment]
> > Generally, i'd say: If you can't teach anything and the list is empty, don't display
> 
> So, something like:
> 
> ```
>     @Override
>     public boolean isPageVisible()
>     {
>         return isVisible && (isRecipeAlterationAllowed() || getRecipes().size() > 0);
>     }
> ```

Was thinking of actually putting that into isVisible on the serverSide already.

**Review by uecasm — APPROVED** [review]

**Review by someaddons — APPROVED** [review]

**Pull request merged**